### ISSUE TITLE
Bug report: Unable to uninstall Hipchat

### ISSUE THREAD

**Issue by manfrin** [caskroom/homebrew-cask#18264]
(Remember to run `brew update; brew cleanup; brew cask cleanup` before reporting your issue. It usually fixes things!)

### Description of issue
Brew cask lists `hipchat`, hipchat is on my system, but cannot uninstall hipchat through cask. 

### Output of `brew cask uninstall hipchat --verbose`

```
➜  ~  brew cask uninstall hipchat
Error: hipchat is not installed
➜  ~  brew list
autoconf	freetype	jpeg		libtiff		mysql		openssl		rbenv		xz
cmake		imagemagick	libpng		libtool		node		pkg-config	ruby-build
➜  ~  brew cask list
alfred	    caffeine	dropbox	    evernote	hipchat	    iterm2	java	    sequel-pro	spotify	    vagrant	virtualbox  vlc
➜  ~  brew uninstall hipchat
Error: No such keg: /usr/local/Cellar/hipchat
➜  ~  brew cask uninstall hipchat --verbose
Error: hipchat is not installed
Error: Kernel.exit
➜  ~  brew cask list
alfred	    caffeine	dropbox	    evernote	hipchat	    iterm2	java	    sequel-pro	spotify	    vagrant	virtualbox  vlc
➜  ~  brew cask cleanup
==> Removing dead symlinks
Nothing to do
==> Removing cached downloads
Nothing to do
➜  ~  brew doctor
Your system is ready to brew.
```

### Output of `brew doctor`

```
Your system is ready to brew.
```

### Output of `brew cask doctor`

```
➜  ~  brew cask doctor
==> OS X Release:
10.11
==> OS X Release with Patchlevel:
10.11.2
==> Hardware Architecture:
intel-64
==> Ruby Version:
2.0.0-p645
==> Ruby Path:
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
==> Homebrew Version:
Homebrew 0.9.5 (git revision 470ea; last commit 2016-02-04)
==> Homebrew Executable Path:
/usr/local/bin/brew
==> Homebrew Cellar Path:
/usr/local/Cellar
==> Homebrew Repository Path:
/usr/local
==> Homebrew Origin:
https://github.com/Homebrew/homebrew
==> Homebrew-cask Version:
0.60.0 (git revision 23c184; last commit 22 minutes ago)
==> Homebrew-cask Install Location:
<NONE>
==> Homebrew-cask Staging Location:
/opt/homebrew-cask/Caskroom
==> Homebrew-cask Cached Downloads:
/Library/Caches/Homebrew
/Library/Caches/Homebrew/Casks
0 files, 0B
==> Homebrew-cask Default Tap Path:
/usr/local/Library/Taps/caskroom/homebrew-cask
==> Homebrew-cask Alternate Cask Taps:
<NONE>
==> Homebrew-cask Default Tap Cask Count:
2967
==> Contents of $LOAD_PATH:
/usr/local/Library/Taps/caskroom/homebrew-cask/lib
/usr/local/Library/Homebrew
/Library/Ruby/Site/2.0.0
/Library/Ruby/Site/2.0.0/x86_64-darwin15
/Library/Ruby/Site/2.0.0/universal-darwin15
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/x86_64-darwin15
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/universal-darwin15
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/x86_64-darwin15
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/universal-darwin15
==> Contents of $RUBYLIB Environment Variable:
<NONE>
==> Contents of $RUBYOPT Environment Variable:
<NONE>
==> Contents of $RUBYPATH Environment Variable:
<NONE>
==> Contents of $RBENV_VERSION Environment Variable:
<NONE>
==> Contents of $CHRUBY_VERSION Environment Variable:
<NONE>
==> Contents of $GEM_HOME Environment Variable:
<NONE>
==> Contents of $GEM_PATH Environment Variable:
<NONE>
==> Contents of $BUNDLE_PATH Environment Variable:
<NONE>
==> Contents of $PATH Environment Variable:
PATH="/usr/local/bin:/usr/local/var/rbenv/shims:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/Library/Taps/caskroom/homebrew-cask/cmd:/usr/local/Library/ENV/scm"
==> Contents of $SHELL Environment Variable:
SHELL="/bin/zsh"
==> Contents of Locale Environment Variables:
LANG="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
==> Running As Privileged User:
No
```

I have done everything I can think of/find on other issues: upgraded, reinstalled brew-cask, cleaned up, doctored, audited, cleaned up again, + everything listed on the bug report guidelines. Cask shows hipchat is there, I can run hipchat, but I cannot seem to uninstall it (version is out of date, and hipchats own 'check for updates' isn't seeing it, so I just want to remove it and reinstall from their download page). 

My only guess is that maybe the hipchat self-updater changed some path, so cask's symlink is still there but it points to nothing?

**Comment by adidalal** [comment]
`brew cask uninstall hipchat --force` will kill any symlink remnants. You can also try dragging  the app into AppCleaner or a similar program to clean up.

Will take a look at the cask shortly

**Comment by adidalal** [comment]
Updated cask, maybe it will make a difference? Please report back with results from the above.

**Comment by manfrin** [comment]
`--force` removed the symlink. Hipchat still here, but I imagine it's just detached and I can uninstall through OSX.

**Comment by adidalal** [comment]
Correct, moving the .app to the trash or cleaning up with AppCleaner will clean up remnants. 

`brew install hipchat` *shouldn't* cause further issues, as we've added in a flag to not overwrite the program's updates, but of course up to you if you prefer to install directly from the website.

**Comment by jawshooah** [comment]
This is expected (albeit broken) behavior. You need to add the `--force` flag to uninstall old versions.

> Hipchat still here

Do you mean that the app bundle remains in your caskroom, or that you had the application open and it did not automatically close? The latter is expected; the OS has the application loaded in memory and will continue to run it even if the bundle has been deleted. The former would be a bug.

**Comment by manfrin** [comment]
@jawshooah I mean it was closed, but uninstalling via cask left it as is (and openable afterwards). 

Installing again seems to have overwritten correctly, as Hipchat is now up to date (using `brew cask install hipchat`). 

In addition, I noticed that in the course of updating, `alfred` seemed to get detached (judging from it not finding hipchat). `brew cask install alfred` appeared to update (and fix the detachment) and typing in `hipchat` in alfred found the newly installed Hipchat.

I think it's just that I had not updated in a while (since before the december switch), and some artifacts we left over that `cleanup` couldn't see/account for. Doing simple `install`s of detached/lost things seems to be an appropriate fix. 

I don't think this is really something that can be fixed; I probably just had very old code that caused issues.
